overlord/devicestate/firstboot.go

```diff
@@ -337,8 +339,8 @@ func importAssertionsFromSeed(st *state.State, sysLabel string) (seed.Seed, erro
 	// collect and
 	// set device,model from the model assertion
 	deviceSeed, err := loadDeviceSeed(st, sysLabel)
-	if err == seed.ErrNoAssertions && release.OnClassic {
-		// on classic seeding is optional
+	if err == seed.ErrNoAssertions && !hasModeEnv && release.OnClassic {
+		// if classic boot seeding is optional
 		// set the fallback model
 		err := setClassicFallbackModel(st, device)
 		if err != nil {
```

Comment:
there is code later in this function that does:

release.OnClassic != classicModel

that will not be correct in modes != run for a classic with modes system, maybe not to fix here yet but probably worth adding at least a comment there

Reply:
Added comment

Reply:
thanks